### DISCUSSION TITLE
CocSearch in parent directory or current workspace

### DISCUSSION THREAD

**Discussion by GodefroyClair** [neoclide/coc.nvim#4589] [General]
From what I understand, if I do a simple `:CocSearch some_pattern`, the search will be done on directory of the current file in buffer and its subdirectories.
What if I want to search on the parent directory? I tried various way based on the documentation but nothing would work.
Also, would there be a way to search on current _workspace_?

**Comment by fannheyward** [comment]
`CocSearch` uses `rg` to search, in `cwd`. ~There's no way to change the path to search in workspace for now.~

You can add workspace root to make `rg` search from, `:CocSearch some_pattern -- /full/path/to/your/workspace`

Note the must-have `-- `.

**Reply by GodefroyClair** [reply]
Thanks great ! Would there be a way to add that to the documentation ?

**Reply by fannheyward** [reply]
I'll add this to document.

**Reply by GodefroyClair** [reply]
after trying I found something weird...
Imagine the following dir substructure :
foo
├── bar
 │        └── baz.c
└── qux.c
with the word "hello" in baz.c and qux.c

If I do
`vim  foo/bar/baz.c`
and then
`:CocSearch hello`
I get :
```
Save current buffer to make changes
Files: 1 Matches: 1 Duration: 43ms

baz.c 1:2
hello
```

But if I do:
`:CocSearch hello -- ..`
the result is:
```
Save current buffer to make changes
Files: 3 Matches: 3 Duration: 37ms

/Users/xxxxx/test/foo/qux.c 1:1
hello
　
baz.c 1:2
hello

baz.c 1:2
hello
```
Hence it seems like I have a duplicate : it seems like it returns the result of both requesting the current directory and the directory after  `--`

Whereas, from command line, there are no duplicate:
```
$ cd foo/bar
$  rg -i hello
baz.c
1:hello
$ rg -i hello -- ..
../bar/baz.c
1:hello

../qux.c
1:hello
```
